engine/runtime-compiler/src/main/java/org/enso/compiler/pass/resolve/MethodDefinitions.java

```diff
@@ -0,0 +1,338 @@
+    var newDefs =
+        ir.bindings()
+            .map(
+                def -> {
+                  if (def instanceof Method method) {
+                    var methodRef = method.methodReference();
+                    Option<Name> resolvedTypeRef =
+                        methodRef.typePointer().map(tp -> resolveType(tp, availableSymbolsMap));
+                    var resolvedMethodRef = methodRef.copyWithTypePointer(resolvedTypeRef);
+
+                    return switch (method) {
+                      case Method.Explicit explicitMethod -> {
+                        var isStatic = computeIsStatic(explicitMethod.body());
+                        var resolvedMethod =
+                            explicitMethod.copy(
+                                resolvedMethodRef,
+                                explicitMethod.body(),
+                                isStatic,
+                                explicitMethod.isPrivate(),
+                                explicitMethod.isStaticWrapperForInstanceMethod(),
+                                explicitMethod.location(),
+                                explicitMethod.passData(),
+                                explicitMethod.diagnostics(),
+                                explicitMethod.id());
+                        yield resolvedMethod;
+                      }
+                      case Method.Conversion conversionMethod -> {
+                        var sourceTypeExpr = conversionMethod.sourceTypeName();
+                        Name resolvedName =
+                            switch (sourceTypeExpr) {
+                              case Name name -> resolveType(name, availableSymbolsMap);
+                              default -> new Conversion(
+                                  sourceTypeExpr,
+                                  UnsupportedSourceType$.MODULE$,
+                                  new MetadataStorage());
+                            };
+                        var resolvedMethod =
+                            conversionMethod.copy(
+                                resolvedMethodRef,
+                                resolvedName,
+                                conversionMethod.body(),
+                                conversionMethod.location(),
+                                conversionMethod.passData(),
+                                conversionMethod.diagnostics(),
+                                conversionMethod.id());
+                        yield resolvedMethod;
+                      }
+                      default -> throw new CompilerError(
+                          "Unexpected method type in MethodDefinitions pass.");
+                    };
+                  } else {
+                    return def;
+                  }
+                });
+
+    java.util.List<Definition> withStaticAliases = new ArrayList<>();
+    for (var def : CollectionConverters.asJava(newDefs)) {
```

Comment:
I guess the above `.map` body and the `for` loop body could easily be merged together into one big `for` loop.